algolia/internal/opt/extra_options.go

```diff
@@ -0,0 +1,28 @@
+// Code generated by go generate. DO NOT EDIT.
+
+package opt
+
+import (
+	"github.com/algolia/algoliasearch-client-go/algolia/opt"
+)
+
+// ExtractExtraOptions returns the first found ExtraOptionsOption from the
+// given variadic arguments or nil otherwise. If multiple options are found, the
+// inner maps are merged.
+func ExtractExtraOptions(opts ...interface{}) *opt.ExtraOptionsOption {
+	merged := make(map[string]interface{})
+
+	for _, o := range opts {
+		if v, ok := o.(*opt.ExtraOptionsOption); ok {
+			for key, value := range v.Get() {
+				merged[key] = value
```

Comment:
what if `key` was already set? should we return an error?

Reply:
For other `opt.Extract*` functions in the client which operate on maps, we merge and override, this is the behavior we've choosen.

For instance, if you're doing:
```
index.Search(
	"",
	opt.ExtraHeaders(map[string]string{"My-Header": "Value 1"}),
	opt.ExtraHeaders(map[string]string{"My-Header": "Value 2"}),
)
```

You're gonna endup sending `Value 2` as an extra header.

**TL;DR** Consistency with other `Extract*` on maps.

Reply:
👍